### PR TITLE
Add comprehensive tests, enforce positive mass invariant, and improve simulation robustness

### PR DESCRIPTION
**Pull request by MaksymPrymierov**
This PR significantly improves the reliability and test coverage of the N-body simulation project.

- Added comprehensive unit and integration tests across gravity, particle, nbodysystem, cli, and gui/engine modules.
- Enforced physical invariants: particle mass must always be positive.
- Improved handling of edge cases such as overlapping particles and zero/negative mass.
- Added real-world validation tests (Earth–Moon, Sun–Earth systems).
- Expanded dependencies (tokio features, tempfile) to support async and file-based tests.

Overall, these changes strengthen correctness, robustness, and confidence in the simulation’s physics and I/O behaviors.

### CODE REVIEW

File: src/types/particle.rs [hunk 1]
```diff
@@ -27,8 +27,12 @@ impl Particle {
         pos: Vector3<f64>,
         velocity: Vector3<f64>,
         acceleration: Vector3<f64>,
-        mass: f64,
+        mut mass: f64,
     ) -> Self {
+        if mass < 0.0 {
+            println!("mass must be > 0");
```
[the remark this answers is not in the document]
Comment (MaksymPrymierov) [reply src/types/particle.rs]: Please check updates
